numpy/core/src/multiarray/ctors.c

```diff
@@ -1806,7 +1812,29 @@ PyArray_GetArrayParamsFromObject(PyObject *op,
             return 0;
         }
         /* If object arrays are forced */
-        if (is_object) {
+        if (is_object != DISCOVERED_OK) {
+            static PyObject * visibleDeprecationWarning = NULL;
+            if (visibleDeprecationWarning == NULL) {
```

Comment:
`npy_cache_import` includes this if (and is guaranteed to be inlined)

Reply:
fixing